runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3156,6 +3156,62 @@ remoteCompilationEnd(
    return relocatedMetaData;
    }
 
+void
+updateCompThreadActivationPolicy(TR::CompilationInfoPerThreadBase *compInfoPT, JITServer::ServerMemoryState nextMemoryState)
+   {
+   // When the server starts running low on memory, clients need to reduce load on the server.
+   // This is achieved by suspending compilation threads once low server memory is detected.
+   //
+   // Under normal conditions, a client uses AGGRESSIVE activation policy, starting new compilation
+   // threads at lower thresholds then non-JITServer.
+   //
+   // Once the server reaches LOW threshold, the client enters MAINTAIN activation policy,
+   // it keeps the current threads running but is not allowed to start new ones.
+   //
+   // If the situation deteriorates further and the server reaches VERY_LOW threshold,
+   // client begins SUSPEND activation policy, disabling all but one compilation threads.
+   //
+   // If at some point the situation improves and the server returns to NORMAL memory state,
+   // then clients using MAINTAIN policy will begin SUBDUE policy, which allows them to start/resume
+   // compilation threads, but at a much higher activation threshold to avoid overwhelming the server again.
+   //
+   auto *compInfo = compInfoPT->getCompilationInfo();
+   JITServer::CompThreadActivationPolicy curPolicy = compInfo->getCompThreadActivationPolicy();
+   if (nextMemoryState == JITServer::ServerMemoryState::VERY_LOW)
+      {
+      compInfo->setCompThreadActivationPolicy(JITServer::CompThreadActivationPolicy::SUSPEND);
+      }
+   else if (nextMemoryState == JITServer::ServerMemoryState::LOW)
+      {
+      compInfo->setCompThreadActivationPolicy(JITServer::CompThreadActivationPolicy::MAINTAIN);
+      }
+   else // ServerMemoryState::NORMAL 
+      {
+      if (curPolicy <= JITServer::CompThreadActivationPolicy::MAINTAIN)
+         {
+         // It's expected that client will transition from MAINTAIN to SUBDUE,
+         // but it's possible to move from SUUSPEND to SUBDUE, if there is a sudden
+         // increase in available memory on the server or a large delay in-between updates
+         compInfo->setCompThreadActivationPolicy(JITServer::CompThreadActivationPolicy::SUBDUE);
+         }
+      else
+         {
+         // For updating the state of the previous policy
+         compInfo->setCompThreadActivationPolicy(curPolicy);
```

Comment:
Since in this case we maintain the current policy, we don't need to set it. This reduces the likelihood of a synchronization issue.

Reply:
Right, I forgot to remove it from the version where I was storing previous policy.